packages/astro/src/assets/fonts/logic/dedupe-font-faces.ts

```diff
@@ -0,0 +1,30 @@
+import type * as unifont from 'unifont';
+import { sortObjectByKey } from '../utils.js';
+
+export function dedupeFontFaces(
+	current: Array<unifont.FontFaceData>,
+	incoming: Array<unifont.FontFaceData>,
+): Array<unifont.FontFaceData> {
+	const result: Array<unifont.FontFaceData> = [...current];
+	for (const font of incoming) {
+		const existing = result.find(({ src: _src, ...rest }) => {
+			const a = JSON.stringify(sortObjectByKey(rest));
+			const { src: __src, ..._rest } = font;
+			const b = JSON.stringify(sortObjectByKey(_rest));
+			return a === b;
+		});
+
+		if (!existing) {
+			result.push(font);
+			continue;
+		}
+
+		const ids = new Set<string>();
+
+		existing.src = [...existing.src, ...font.src].filter((source) => {
+			const id = 'name' in source ? source.name : source.url;
+			return !ids.has(id) && ids.add(id);
+		});
```

Comment:
Nit: I believe it's more memory efficient to filter the current arrays:

```js
function filterSource() {

}

existing.push(...existing.src.filter(filterSource));
existing.push(...font.src.filter(filterSource));
```

Or use `concat` for a functional approach:

```js
existing = existing.concat(...existing.src.filter(filterSource));
existing = existing.concat(...font.src.filter(filterSource));
```